src/CodeGen_LLVM.cpp

```diff
@@ -2613,14 +2598,17 @@ void CodeGen_LLVM::visit(const Call *op) {
         llvm::CallInst *call = builder->CreateCall(base_fn->getFunctionType(), phi, call_args);
         value = call;
     } else if (op->is_intrinsic(Call::prefetch)) {
-        user_assert((op->args.size() == 3) && is_one(op->args[1]))
+        user_assert((op->args.size() == 4) && is_one(op->args[2]))
             << "Only prefetch of 1 cache line is supported.\n";
 
         llvm::Function *prefetch_fn = module->getFunction("_halide_prefetch");
         internal_assert(prefetch_fn);
 
+        const Variable *base = op->args[0].as<Variable>();
+        internal_assert(base && base->type.is_handle());
+
         vector<llvm::Value *> args;
-        args.push_back(codegen(op->args[0]));
+        args.push_back(codegen_buffer_pointer(base->name, op->type, op->args[1]));
```

Comment:
Instead of requiring that args[0] be a variable with the correct name, I think it would be better to add an overload of codegen_buffer_pointer that takes a base address instead of a base name. The current version of codegen_buffer_pointer would be a one-liner that looks up the name in the symbol table and calls the new version.